src/Components/Footer/Footer.tsx

```diff
@@ -0,0 +1,67 @@
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+
+type FooterProps = {
+  todos: Todo[];
+  status: string;
+  setStatus: (status: string) => void;
+};
+export const Footer: React.FC<FooterProps> = ({ todos, status, setStatus }) => {
+  const activeTodoCount = todos.filter(todo => !todo.completed).length;
+  const completedTodoCount = todos.filter(todo => todo.completed).length;
+
+  const handleStatusChange = (newStatus: string) => {
+    setStatus(newStatus);
+  };
+
+  return (
+    <>
+      {/* Hide the footer if there are no todos */}
+      <footer className="todoapp__footer" data-cy="Footer">
+        <span className="todo-count" data-cy="TodosCounter">
+          {`${activeTodoCount} items left`}
+        </span>
+
+        {/* Active link should have the 'selected' class */}
+        <nav className="filter" data-cy="Filter">
+          <a
+            href="#/"
+            className={cn('filter__link', { selected: status === 'all' })}
+            data-cy="FilterLinkAll"
+            onClick={() => handleStatusChange('all')}
```

Comment:
Then you can use it in this place as: 
```suggestion
            onClick={handleStatusChange('all')}
```